lib/iris/tests/unit/util/test_flatten_multidim_coord.py

```diff
@@ -0,0 +1,96 @@
+from iris.coords import DimCoord, AuxCoord
+from iris.util import flatten_multidim_coord, flatten_cube
+
+
+class Test(tests.IrisTest):
+    def test_argument_is_basestring(self):
+        cube_a = stock.simple_2d_w_multidim_coords()
+
+        cube_b = flatten_multidim_coord(cube_a, 'bar')
+        self.assertEqual(cube_b.shape, (12, ))
+
+    def test_argument_is_coord_instance(self):
+        cube_a = stock.simple_3d_w_multidim_coords()
+        coord = cube_a.coord('bar').copy()
+        cube_b = flatten_multidim_coord(cube_a, coord)
+        self.assertEqual(cube_b.dim_coords, (cube_a.coord('wibble'), ))
+        self.assertEqual(cube_b.shape, (2, 12))
+
+    def test_simple_cube_flattened(self):
+        cube_a = stock.simple_2d()
+        cube_b = flatten_cube(cube_a)
+        self.assertEqual(cube_b.dim_coords, tuple())
+        self.assertEqual(cube_b.shape, (12, ))
+
+    def test_oned_dim_coord_flattened(self):
+        cube_a = stock.simple_3d_w_multidim_coords()
+        cube_a.add_dim_coord(DimCoord([0, 1, 2], var_name='blah'), (1,))
+        cube_b = flatten_cube(cube_a, (1, 2))
+        self.assertEqual(cube_b.dim_coords, (cube_a.coord('wibble'), ))
+        self.assertEqual(cube_b.coord('blah', dim_coords=False).shape, (12, ))
+        self.assertEqual(cube_b.shape, (2, 12))
+
+    def test_multiple_oned_dim_coords_flattened(self):
+        cube_a = stock.simple_3d_w_multidim_coords()
+        cube_a.add_dim_coord(DimCoord([0, 1, 2], var_name='blah'), (1,))
+        cube_a.add_dim_coord(DimCoord([0, 1, 2, 3], var_name='blah2'), (2,))
+        cube_b = flatten_cube(cube_a, (1, 2))
+        self.assertEqual(cube_b.dim_coords, (cube_a.coord('wibble'), ))
+        self.assertEqual(cube_b.coord('blah', dim_coords=False).shape, (12, ))
+        self.assertEqual(cube_b.coord('blah2', dim_coords=False).shape, (12,))
+        self.assertEqual(cube_b.shape, (2, 12))
+
+    def test_oned_aux_coord_flattened(self):
+        cube_a = stock.simple_3d_w_multidim_coords()
+        cube_a.add_aux_coord(AuxCoord([0, 1, 2], var_name='blah'), (1,))
+        cube_b = flatten_cube(cube_a, (1, 2))
+        self.assertEqual(cube_b.dim_coords, (cube_a.coord('wibble'), ))
+        self.assertEqual(cube_b.coord('blah', dim_coords=False).shape, (12, ))
+        self.assertEqual(cube_b.shape, (2, 12))
+
+    def test_aux_coords_leading(self):
+        cube_a = stock.simple_3d_w_multidim_coords()
+        # Move the aux coord dims to the front of the cube
+        cube_a.transpose((1, 2, 0))
+        cube_b = flatten_cube(cube_a, (0, 1))
+        self.assertEqual(cube_b.dim_coords, (cube_a.coord('wibble'), ))
+        self.assertEqual(cube_b.shape, (12, 2))
+
+    def test_split_aux_coord_dims(self):
+        cube_a = stock.simple_3d_w_multidim_coords()
+        cube_a.transpose((1, 0, 2))
+        cube_b = flatten_cube(cube_a, (0, 2))
+        self.assertEqual(cube_b.dim_coords, (cube_a.coord('wibble'), ))
+        self.assertEqual(cube_b.shape, (12, 2))
+
```

Comment:
Again, I'm not sure if this test adds any value that hasn't already been added by `test_aux_coords_leading`, so I would suggest removing it.

Reply:
This is useful because there is some special logic for dealing with these types of coordinates